src/main/java/org/checkerframework/languageserver/CFDiagnostic.java

```diff
@@ -9,17 +9,37 @@
  * JSON wrapper to communicate javac diagnostics from {@link JavacWrapper} to {@link CheckExecutor}.
  */
 public class CFDiagnostic implements Diagnostic<Object> {
-
+    /** The file URI. */
     private final String fileUri;
+
+    /** The diagnostics kind. It is one of value in {@link Kind} enum. */
```

Comment:
Why is this a string and not `Kind`?

Reply:
Note how the field is set in the constructor and then converted back to the enum in the getter. This is very odd.
Overall, couldn't this class simply store a reference to the `Diagnostic` object that is passed in the constructor and then delegate all queries to that object? Storing the info in these fields doesn't seem to do anything useful.
Can you think about this and break this out into a separate PR?

Reply:
You are right, this will simplify a lot for the code. I will open a separate  PR for this.

Reply:
Please open an issue for this follow-up work.

Reply:
Opened a issue at #245.